native/src/main/scala/sqs4s/auth/Credentials.scala

```diff
@@ -68,101 +78,122 @@ object Credentials {
     ttl: FiniteDuration = 6.hours,
     refreshBefore: FiniteDuration = 5.minutes
   ): Resource[F, Credentials[F]] =
-    envVar[F] orElse sysProp[F] orElse instanceMetadata[F](
+    envVar[F] orElse sysProp[F] orElse containerMetadata[F](
       client,
       ttl,
       refreshBefore
-    )
-
-  /**
-    * Create a resource of pure value Credential where the access key and secret
-    * key are static
-    *
-    * @param accessKey static access key
-    * @param secretKey static secret key
-    * @tparam F an effect which represents pure value
-    * @return a Resource of TemporarySecurityCredential
-    */
-  def basicResource[F[_]: Applicative](
-    accessKey: String,
-    secretKey: String
-  ): Resource[F, Credentials[F]] =
-    Resource.pure[F, Credentials[F]](instance(BasicCredential(
-      accessKey,
-      secretKey
-    ).pure[F].widen[Credential]))
+    ) orElse instanceMetadata[F](
+      client,
+      ttl,
+      refreshBefore
+    ).handleErrorWith(_ =>
```

Comment:
Technically speaking, this is handling the error of the last one and not of all of them.
The end result should be the same, but I would put the extra parenthesis around all the expression.

I would even do it like this:

```scala
def chain[F[_]: Concurrent: Clock: Timer](
  client: Client[F],
  ttl: FiniteDuration = 6.hours,
  refreshBefore: FiniteDuration = 5.minutes
): Resource[F, Credentials[F]] = {
  val env = envVar[F]
  val sys = sysProp[F]
  val container = containerMetadata[F](client, ttl, refreshBefore)
  val instance = instanceMetadata[F](client, ttl, refreshBefore)

  val tryAllInOrder = env orElse sys orElse container orElse instance

  tryAllInOrder.handleErrorWith { _ =>
    Resource.liftF(Sync[F].raiseError[Credentials[F]](NoValidAuthMethodError))
  }
}
```